File: fundamentos/desafio-js/src/questions/sun-of-deposits.ts
import { IAccount } from "../i-account";

/** A Soma total dos depósitos de todas as agências é: */
export default function sunOfDeposits(...accounts: Array<IAccount>) {
  const total = accounts.reduce((prev, current) => ({ ...prev, balance: prev.balance + current.balance }), { balance: 0 });
  console.info(`A Soma total dos depósitos de todas as agências é: ${total.balance}`);
}

File: fundamentos/desafio-dart/lib/questions/sun_of_deposits.dart
import 'package:desafio_dart/domains/account.dart';

/// A Soma total dos depósitos de todas as agências é:
void sunOfDeposits(List<Account> accounts) {
  final total = accounts.reduce((prev, current) =>
      current.copyWith(balance: (prev?.balance ?? 0) + current.balance));

  print('A Soma total dos depósitos de todas as agências é: ${total.balance}');
}
